### PR TITLE
[master]Support to create swap device with vdisk

### PR DESCRIPTION
**Pull request by trenxiaof**
https://github.ibm.com/zvc/planning/issues/10607

### CODE REVIEW

**Comment by trenxiaof** [comment]
@jichenjc @SeanHQF FYI

File: zvmsdk/smtclient.py [hunk 1]
```diff
@@ -738,43 +738,43 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
...
-                            errmsg = ("%s is great than 2G" % sizeUpper)
-                            raise exception.SDKInvalidInputFormat(msg=errmsg)
-
-                    rd += ' --vdisk %s:%s' % (vd, sizeUpper)
-                    vdisk = disk
+        # If customer uses vdisk to create swap disk, we need
+        # handle the swap disk and non-swap disk seperately,
+        # for the swap only case, customer using boot
+        # from volume but no disk pool provided, non-swap disk is empty
+        swap_list = []
```
Comment (jichenjc) [inline zvmsdk/smtclient.py:745]: I don't know whether we want to do this, but swap is always 1 disk

File: zvmsdk/smtclient.py [hunk 2]
```diff
@@ -738,43 +738,43 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
...
+                   if sizeUnit == 'G':
+                       size = int(sizeUpper[:-1])
+                       if size > 2:
+                           errmsg = ("%s is great than 2G" % sizeUpper)
+                           raise exception.SDKInvalidInputFormat(msg=errmsg)
+                   rd += ' --vdisk %s:%s' % (vd, sizeUpper)
+                   # Remove the vdisk from disk_list, then in the following
+                   # we will only add the remaining disks for the userid
+                   disk_list.pop(i)
+                   swap_list.append(disk)
```
Comment (jichenjc) [inline zvmsdk/smtclient.py:777]: I'd like to see some UT for this, e.g 
- no force MDISK (which means VDISK) , what's the param gonna send to, e.g what's the output of disk_list
- force MDISK, what's the output list
- only swap (BFV case) , will that be different?

File: zvmsdk/smtclient.py (hunk 1 above)
Comment (trenxiaof) [reply zvmsdk/smtclient.py:745]: we don't support multiple swap disks? Originally I though only 1 swap from UI, maybe we can create multiple swap with command line.

Comment (jichenjc) [reply zvmsdk/smtclient.py:745]: ok you are right, seems Linux support multiple swap but openstack only support 1 swap I THink

File: zvmsdk/smtclient.py (hunk 2 above)
Comment (trenxiaof) [reply zvmsdk/smtclient.py:777]: I tested no force Mdisk for the following cases:
1. 1 ephemral + 1 swap:
[{'size': '10g', 'is_boot_disk': True, 'format': 'none', 'disk_pool': 'eckd:icicpl', 'vdev': '0100'},
 {'size': '1g', 'format': 'ext4', 'disk_pool': 'eckd:icicpl', 'vdev': '0101'},
 {'size': '200m', 'format': 'swap', 'disk_pool': 'eckd:icicpl', 'vdev': '0102'}]--->created by vdisk

MDISK 0102 FB-512 V-DISK 614400 MWV
MDISK 0100 3390 902980 14564 CI2101 MR
MDISK 0101 3390 977518 1457 CI2100 MR

BFV with swap only and disk_pool is none:
it creates swap  device by Vdisk

BFV with swap only and disk_pool is not None:
it creates swap  device by Vdisk

Will continue to test the force Mdisk resultes.

**Pull request closed without merging**